### PR TITLE
Adding Android Quickstart

### PR DESCRIPTION
**Pull request by vcampitelli**
There are some things that I wanted to check before we can actually merge this:

- [x] Decide on the name: should it actually be "Java for Android"?
- [x] I've named the sample repo as [`fusionauth-quickstart-android-java`](https://github.com/FusionAuth/fusionauth-quickstart-android-java) but the [Quickstart instructions](https://github.com/FusionAuth/fusionauth-example-template/blob/master/QUICKSTART-INSTRUCTIONS.md#the-tldr) expects `fusionauth-quickstart-LANGUAGE-FRAMEWORK-{api,web}`, so maybe I should append `-native`? I didn't because it seems redundant.
- [x] The Issuer should match the instance URL, so the user should expose the instance with ngrok, update the Issuer and recreate asymmetric keys. Should I add these steps to the guide or create a simple script (probably add a command to the Node CLI) that does all that automatically?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205088227520253
  - https://app.asana.com/0/0/1205273462533594

### CODE REVIEW

**Comment by mooreds** [comment]
> Decide on the name: should it actually be "Java for Android"?

How about "Android Java Quickstart"?

> I've named the sample repo as [fusionauth-quickstart-android-java](https://github.com/FusionAuth/fusionauth-quickstart-android-java) but the [Quickstart instructions](https://github.com/FusionAuth/fusionauth-example-template/blob/master/QUICKSTART-INSTRUCTIONS.md#the-tldr) expects fusionauth-quickstart-LANGUAGE-FRAMEWORK-{api,web}, so maybe I should append -native? I didn't because it seems redundant.

How about fusionauth-quickstart-java-android-native (or -mobile)? Either seem accurate.

> The Issuer should match the instance URL, so the user should expose the instance with ngrok, update the Issuer and recreate asymmetric keys. Should I add these steps to the guide or create a simple script (probably add a command to the Node CLI) that does all that automatically?

Hmmm. Can you have this be a config setting, where you stand up FusionAuth with ngrok, update a settings file someplace, then build the android app?

**Comment by vcampitelli** [comment]
> How about "Android Java Quickstart"?

Done

> How about fusionauth-quickstart-java-android-native (or -mobile)? Either seem accurate.

Done

> Hmmm. Can you have this be a config setting, where you stand up FusionAuth with ngrok, update a settings file someplace, then build the android app?

What do you mean by "config setting"?

**Comment by mooreds** [comment]
> What do you mean by "config setting"?

When I am building an android app, is there a corresponding config file, similar to `.env` for docker or php, where a user could set up the ngrok URL?

So I'm talking about build time configuration, not runtime configuration.

Does that make sense?

**Comment by vcampitelli** [comment]
> When I am building an android app, is there a corresponding config file, similar to .env for docker or php, where a user could set up the ngrok URL?

Yes, it does. That's why I asked if it would be better to write these steps to the quickstart (expose the instance with ngrok, update the Issuer, recreate asymmetric keys, update settings file) or create a simple script to do all that.

**Comment by mooreds** [comment]
>  That's why I asked if it would be better to write these steps to the quickstart (expose the instance with ngrok, update the Issuer, recreate asymmetric keys, update settings file) or create a simple script to do all that.

Ah, okay. Since this won't be done typically (people will probably work through the quickstart once, not multiple times) I think documentation is fine.

**Comment by vcampitelli** [comment]
1. I've pushed an additional step on configuring issuer and regenerating JWT keys.
2. I've also created a `<IconButton>` component to render UI-like buttons, like we had with markdown/asciidocs, and updated `DocsDevREADME` with instructions.
icons
3. As discussed with @mooreds, I didn't recreate the exact same Changebank theme as things aren't that easy with Android, but I used the same color, texts and functionality.

**Review by ColinFrick — CHANGES_REQUESTED** [review]
After completing the tutorial the app didn't start. I had to remove following files first:
- `app/java/io/fusionauth/app/BrowserSelectionAdapter.java`
- `app/res/layout/browser_selector_layout.xml`

CleanShot 2023-09-06 at 09 19 06@2x

File: astro/src/content/quickstarts/quickstart-android-java.mdx [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+<Aside type="note">
+    If you ever want to reset the FusionAuth system, delete the volumes created by docker compose by executing `docker compose down -v`, then re-run `docker compose up -d`.
+</Aside>
+
+FusionAuth will be initially configured with these settings:
+
+* Your client Id is `e9fdb985-9173-4e01-9d73-ac2d60d1dc8e`.
+* Your client secret is `super-secret-secret-that-should-be-regenerated-for-production`.
+* Your example username is `richard@example.com` and the password is `password`.
```
[the remark this answers is not in the document]
Comment (vcampitelli) [reply astro/src/content/quickstarts/quickstart-android-java.mdx:80]: Sorry, I missed a `push` with the right kickstart and CSS files. Done now.

**Comment by mooreds** [comment]
@vcampitelli is this ready for @ColinFrick to re-review?

**Comment by vcampitelli** [comment]
Yup, just pushed the kickstart file that was missing.

Can you please review it again @ColinFrick ?

**Review by ColinFrick — APPROVED** [review]
@vcampitelli Quick start looks good. I have found a typo in the kickstart.json.

See https://github.com/FusionAuth/fusionauth-quickstart-java-android-native/commit/8a38be203ef670ae586115ce9e1ad3d4d35e8940#r127101206

**Comment by vcampitelli** [comment]
Done! Thanks @ColinFrick 

@mooreds this is now ready for you to review

**Review by mooreds — APPROVED** [review]

**Pull request merged**